e2e-tests/src/main/java/com/dnastack/ga4gh/dataconnect/adapter/TablesRegistrySchemaTest.java

```diff
@@ -3,93 +3,54 @@
 import com.dnastack.ga4gh.dataconnect.adapter.test.model.DataModel;
 import com.dnastack.ga4gh.dataconnect.adapter.test.model.Table;
 import com.dnastack.ga4gh.dataconnect.adapter.test.model.TableInfo;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
 import org.assertj.core.api.Assertions;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import java.io.IOException;
-import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import static org.hamcrest.CoreMatchers.not;
 import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
+@Slf4j
+@EnabledIfEnvironmentVariable(named = "E2E_INDEXING_SERVICE_ENABLED", matches = "true", disabledReason = "This app doesn't have indexing-service properties configured.")
```

Comment:
Technically this test should be predicated on the existence of [ga4gh-tables-registry](https://github.com/DNAstack/ga4gh-tables-registry) configuration... something we don't use anymore... meaning this test would always be disabled for the foreseeable future.

Come to think of it, these e2e tests might not need to get down to that level of detail. All they actually care about is "does this table have this schema," right? They don't need to have an opinion about whether the table metadata is coming from ga4gh-tables-registry, indexing-service, or collection-service.

If I understand correctly, perhaps we should merge this class and `SchemaTestWithLibraryTest` into one `TableJsonSchemaTest`? Or even eliminate both these classes and add the ability to verify a JSON Schema (expectedDataModel) to the mainstream table tests?

WDYT?

Reply:
Just to summarize: 
- `SchemaTestWithLibraryTest` has 1 test that just creates a table with custom json schema (`.jsonSchema(objectMapper.writeValueAsString(Map.of("$comment", "This is the custom schema from library")))`), and ensures that the custom schema is shown when it queries data-connect-trino for the table info. 
- `[TablesRegistrySchemaTest` with the help of env variables also queries data-connect-trino for:
   -  the table info and checks that the json schema returned matches what it expected
   - the table data and checks that some data is returned (i.e. not null) and the json schema matches what it expected 

So based on that, I'm down to consolidate both test classes into `DataConnectE2eTest` by adding a test that creates a table with a custom json schema and ensures that:
   - the custom schema is shown when it queries data-connect-trino for the table info 
   - the table data and checks that some data is returned (i.e. not null) and the json schema matches what it expected 

The only thing I'm not sure of is, do we need/want to specifically test for Ian Fore's json data model? We could have the test so that if it finds env vars of the pattern - `^E2E_TRS_([A-Za-z\\d]+)_TABLE_NAME$` then it just uses that, else it will create the table itself. However since the test will already be checking that custom schemas are returned, idk if its worth the added complexity. Wdyt?

Reply:
With [the latest commit](https://github.com/DNAstack/data-connect-trino/pull/15/commits/7216997584a4a3279cce978b57a99f1caf727fbe) I've moved the `SchemaTestWithLibraryTest` and the custom schema check from `TablesRegistrySchemaTest`  into `DataConnectE2eTest`

Reply:
Thanks for the summary. I was getting confused between the data-connect-trino e2e tests and the trino-service e2e tests. Now I see we didn't have any pre-existing parameterized tests at the data-connect-trino level as we do in trino-service.

I think we should be able to test for Ian Fore's schemas specifically since he does rely on that data being there and it's vulnerable to getting wiped out, eg by someone removing the data source from Publisher because they don't know what it's there for. So for that reason, I think it makes sense to consolidate what you already did into a single test class which accepts environment variable patters to verify existing tables and schemas are set up as we expect.

The one new parameterized test shouldn't have `TRS` in the names of the environment variables, since that's a call back to the past when we implemented custom schemas with the ga4gh-tables-registry-service. I think the test should only care that the tables we care about exist, and they have the expected JSON Schemas. The tests shouldn't care how those tables came to exist or have those specific schemas. We can reinforce this by choosing env variable names that talk about tables and their schemas, but not about how they came to exist.

Reply:
Okay so `SchemaTestWithLibraryTest` and `TablesRegistrySchemaTest` have been consolidated into two tests in `DataConnectE2eTest`:
- the `getTableInfo_should_returnCustomSchema_from_indexingService` test, which creates a library entry with a custom schema for a table (created as part of the class setup) and checks that the custom schema is returned when calling `/info` in DCT
- the `getTableInfoAndData_should_returnExpectedDataModel` test, which is run if there are env vars matching the pattern `^E2E_([A-Za-z\\d]+)_EXPECTED_DATA_MODEL$`. It then calls the DCT `/info` and `data` endpoints to ensure the data model matches what it expected. If no matching env vars are found then that test is disabled.

Test run against DCT-P in staging:
![image](https://user-images.githubusercontent.com/4998821/179557185-c53d44ff-3159-429c-8b18-92b8bf2f1c29.png)